File: Examples/ClassDiagram/Swift/MediaItem.swift
import Foundation

/// Shared base for every item in the library.
open class MediaItem: Playable {
    public let title: String
    public let duration: TimeInterval
    public let genre: Genre

    public init(title: String, duration: TimeInterval, genre: Genre) {
        self.title = title
        self.duration = duration
        self.genre = genre
    }

    open func play() {
        print("Playing \(title)…")
    }
}

File: Examples/ClassDiagram/Python/media_item.py
from playable import Genre, Playable


class MediaItem(Playable):
    """Shared base for every item in the library."""

    def __init__(self, title: str, duration: float, genre: Genre):
        self.title: str = title
        self.duration: float = duration
        self.genre: Genre = genre

    def play(self) -> None:
        print(f"Playing {self.title}…")
